### ISSUE TITLE
Add `setuptools` and/or `wheel` to `requirements/host` when used

### ISSUE THREAD

**Issue by jakirkham** [regro/cf-scripts#2888]
In preparation for changes in `pip`'s packaging as discussed in issue ( https://github.com/conda-forge/pip-feedstock/issues/124 ), it would be good if we could detect the usage of `setuptools` and/or `wheel` in Python packages and add them to `requirements/host` if they are needed

**Comment by jakirkham** [comment]
cc @beckermr

**Comment by beckermr** [comment]
yes we can add a mini-migrator but we'll need to figure out a decent way to understand what we need to add - it is not clear to me when we need setuptools-only vs both setuptools+wheel.

**Comment by jakirkham** [comment]
To the extent that Python packages declare these dependencies in their `pyproject.toml`'s `build-system.requires`, [`grayskull` can pick them up and add them]( https://github.com/conda/grayskull/blob/32a7e3905e1d309a47a616fd26c659e721214fab/grayskull/strategy/py_toml.py#L354 )

Is this something that `depfinder` checks?

**Comment by beckermr** [comment]
`depfinder` only detects imports and so would miss this in some cases.

**Comment by jakirkham** [comment]
Can you please share some examples? Identifying the edge cases will help us come up with a better solution

**Comment by beckermr** [comment]
Oh right sorry. Here are the ones I am thinking of. None of them mean we cannot do this, but it will require thought+code.

 - grayskull fails for w/e reason
 - the recipe uses flit, hatchling, meson etc. as a build backend
 - the project only uses a pyproject.toml and so `depfinder` won't check anything since it only looks at python imports
 - there are surely multioutput recipes with odd edge cases

**Comment by jakirkham** [comment]
As an interesting example

`lda` gets flagged as needing `setuptools`: https://github.com/conda-forge/lda-feedstock/pull/12

AFAICT it is only used in [`lda`'s `cibuildwheels` tests]( https://github.com/lda-project/lda/blob/c63e8d7709a3271d383be3d80cb15cd53f9a6451/pyproject.toml#L66 )

**Comment by beckermr** [comment]
Even more interestingly, setuptools is already in host in that recipe. :/

**Comment by beckermr** [comment]
Ahhhh. The bot flagging that is not related to depfinder or grayskull but is because it detected that package in host.

**Comment by jakirkham** [comment]
> To the extent that Python packages declare these dependencies in their `pyproject.toml`'s `build-system.requires`, [`grayskull` can pick them up and add them](https://github.com/conda/grayskull/blob/32a7e3905e1d309a47a616fd26c659e721214fab/grayskull/strategy/py_toml.py#L354)

However it looks like the bot won't apply Grayskull's analysis as part of the update

For example: https://github.com/conda-forge/webcolors-feedstock/pull/29#discussion_r1870463516

**Comment by beckermr** [comment]
I think grayskull does run only maybe to prevent bad updates.

**Comment by jakirkham** [comment]
The feedstock is configured to [apply Grayskull updates]( https://github.com/conda-forge/webcolors-feedstock/blob/10e471cec99892f88bc78f6c371cbb5ba2b43fdf/conda-forge.yml#L3 ) (unless it is misconfigured?)

So would have thought there would have been an update in `requirements/host` in this case, which didn't occur

**Comment by beckermr** [comment]
Right I am saying the bot actively excludes host updates.

**Comment by jakirkham** [comment]
Could we start including them? Or was there some issue with `host` dependencies in the past?

**Comment by beckermr** [comment]
The issue has been that host sections tend to have a lot of special things going on that grayskull does not or cannot do properly. Hence we've avoided it IIRC.
